src/snowflake/cli/plugins/rpc/manager.py

```diff
@@ -0,0 +1,124 @@
+from snowflake.cli.api.sql_execution import SqlExecutionMixin
+from snowflake.snowpark import Session
+
+MAX_WORKERS = 5
+
+
+class ConnectionParams(TypedDict):
+    sessionToken: Optional[str]
+    masterToken: Optional[str]
+    account: Optional[str]
+    host: Optional[str]
+    port: Optional[str]
+    protocol: Optional[str]
+
+
+class LSPPluginContext:
+    """
+    Contains context for LSP Plugins
+    """
+
+    def __init__(self) -> None:
+        self._connection = self.create_connection()
+
+    def get_connection(self):
+        return self._connection
+
+    def update_connection(self, params: ConnectionParams):
+        self._connection = self._create_connection(params)
+
+    def _create_connection(self, params: ConnectionParams):
+        session_token = params["sessionToken"]
+        master_token = params["masterToken"]
+        account = params["account"]
+        host = params["host"]
+        port = params["port"]
+        protocol = params["protocol"]
+        if session_token and master_token:
+            try:
+                conn = connector.connect(
+                    account=account,
+                    host=host,
+                    port=port,
+                    protocol=protocol,
+                    session_token=session_token,
+                    master_token=master_token,
+                    server_session_keep_alive=True,
+                )
+            except connector.ProgrammingError as e:
+                error_message = str(e)
+                # "251005: User is empty" indicates the session token parameter doesn't exist on the current snowflake-python-connector version
+                if "251005" in error_message:
+                    raise connector.ProgrammingError(
+                        "Failed to use the active connection from the Snowflake extension. Requires snowflake-python-connector version 3.7.0 or higher."
+                    )
+                else:
+                    raise
+            return conn
+        else:
+            conn = connector.connect()
+            return conn
+
+    def create_connection(self) -> Session:
+        session_token = os.environ.get("SESSION_TOKEN")
+        master_token = os.environ.get("MASTER_TOKEN")
+        account = os.environ.get("SF_ACCOUNT")
+        host = os.environ.get("SF_HOST")
+        port = os.environ.get("SF_PORT")
+        protocol = os.environ.get("SF_PROTOCOL")
+
+        params: ConnectionParams = {
+            "sessionToken": session_token,
+            "masterToken": master_token,
+            "account": account,
+            "host": host,
+            "port": port,
+            "protocol": protocol,
+        }
+
+        return self._create_connection(params)
+
+
+class RpcManager(SqlExecutionMixin):
+    """
+    Base class LSP Controller
+    """
+
+    def __init__(self):
+        super().__init__()
+        server = LanguageServer(
+            name="lsp_controller", version="v0.0.1", max_workers=MAX_WORKERS
```

Comment:
I'm really worried in how many places we have version strings.

Reply:
Okay, I think it works out if we have a single version number for the lsp plugin, since that's what from pyglas gets passed into ServerInfo to the client during init.